api/client.go

```diff
@@ -487,6 +463,41 @@ func NewClient(c *Config) (*Client, error) {
 	return client, nil
 }
 
+// ParseAddress transforms the provided address into a url.URL and handles
+// the case of Unix domain sockets by setting the DialContext in the
+// configuration's HttpClient.Transport.
+func ParseAddress(address string, config *Config) (*url.URL, error) {
+	u, err := url.Parse(address)
+	if err != nil {
+		return nil, err
+	}
+
+	if strings.HasPrefix(address, "unix://") {
+		socket := strings.TrimPrefix(address, "unix://")
+		transport := config.HttpClient.Transport.(*http.Transport)
+		transport.DialContext = func(context.Context, string, string) (net.Conn, error) {
+			return net.Dial("unix", socket)
+		}
+
+		// Since the address points to a unix domain socket, the scheme in the
+		// *URL would be set to `unix`. The *URL in the client is expected to
+		// be pointing to the protocol used in the application layer and not to
+		// the transport layer. Hence, setting the fields accordingly.
+		u.Scheme = "http"
+		u.Host = socket
+		u.Path = ""
+	} else {
+		transport := config.HttpClient.Transport.(*http.Transport)
+		transport.DialContext = (&net.Dialer{
```

Comment:
This doesn't look quite like how we setup the transport in DefaultConfig.

Reply:
I went deeper than DefaultConfig and peeked into cleanhttp.DefaultPooledTransport (DefaultConfig uses cleanhttp.DefaultPooledClient, which uses cleanhttp.DefaultPooledTransport).  In general, I don't really like this, since I'm getting way down into the implementation details of cleanhttp.  However, I don't want to squash any of the other unaffected fields of the Transport.